pkg/generators/backend/cron_deployment.go

```diff
@@ -0,0 +1,95 @@
+package backend
+
+import (
+	"fmt"
+
+	saasv1alpha1 "github.com/3scale/saas-operator/api/v1alpha1"
+	"github.com/3scale/saas-operator/pkg/basereconciler"
+	"github.com/3scale/saas-operator/pkg/generators/backend/config"
+	"github.com/3scale/saas-operator/pkg/generators/common_blocks/pod"
+	"github.com/3scale/saas-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Deployment returns a basereconciler.GeneratorFunction funtion that will return a Deployment
+// resource when called
+func (gen *CronGenerator) Deployment(hashErrorMonitoring string) basereconciler.GeneratorFunction {
+
+	return func() client.Object {
+
+		dep := &appsv1.Deployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Deployment",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      gen.GetComponent(),
+				Namespace: gen.Namespace,
+				Labels:    gen.GetLabels(),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: gen.CronSpec.Replicas,
+				Selector: gen.Selector(),
+				Strategy: appsv1.DeploymentStrategy{
+					Type: appsv1.RollingUpdateDeploymentStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateDeployment{
+						MaxUnavailable: util.IntStrPtr(intstr.FromInt(0)),
+						MaxSurge:       util.IntStrPtr(intstr.FromInt(1)),
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: gen.LabelsWithSelector(),
+						Annotations: map[string]string{
+							saasv1alpha1.RolloutTriggerAnnotationKeyPrefix + config.ErrorMonitoringSecretName: hashErrorMonitoring,
+						},
+					},
+					Spec: corev1.PodSpec{
+						ImagePullSecrets: func() []corev1.LocalObjectReference {
+							if gen.Image.PullSecretName != nil {
+								return []corev1.LocalObjectReference{{Name: *gen.Image.PullSecretName}}
+							}
+							return nil
+						}(),
+						Containers: []corev1.Container{
+							{
+								Name:  gen.GetComponent(),
+								Image: fmt.Sprintf("%s:%s", *gen.Image.Name, *gen.Image.Tag),
+								Args: func() (args []string) {
+									args = []string{
+										"backend-cron",
+									}
+									return
+								}(),
```

Comment:
Yoy don't need to use a function here, you can just use
```suggestion
								Args: []string{"backend-cron"},
```

Reply:
Yes I know, I left it prepared as a function in case it is needed in the future to have optional args like backend-listener (with `if` statements), but yes, we can skip it and add the function in the future if needed

Reply:
So, knowing that, what do you consider better? simplify it? o leave it inside a function?